client/my-sites/email/form/mailboxes/components/mailbox-form-wrapper/index.tsx

```diff
@@ -1,35 +1,35 @@
 import { useRtl } from 'i18n-calypso';
+import { PropsWithChildren } from 'react';
 import { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
 import { MailboxField } from 'calypso/my-sites/email/form/mailboxes/components/mailbox-field';
 import {
 	EmailProvider,
 	GoogleMailboxFormFields,
 	MailboxFormFieldBase,
+	MailboxFormFields,
 	TitanMailboxFormFields,
 } from 'calypso/my-sites/email/form/mailboxes/types';
 
 interface MailboxFormWrapperProps {
+	index: number;
+	isAutoFocusEnabled: boolean;
 	mailbox: MailboxForm< EmailProvider >;
 	onFieldValueChanged?: ( field: MailboxFormFieldBase< string > ) => void;
 }
 
-const MailboxFormWrapper = ( {
-	children,
-	mailbox,
-	onFieldValueChanged = () => undefined,
-}: MailboxFormWrapperProps & {
-	children?: JSX.Element | false;
-} ): JSX.Element => {
-	const isRtl = useRtl();
-
-	const formFields = mailbox.formFields;
-	const domainAffix = {
-		[ isRtl ? 'textInputPrefix' : 'textInputSuffix' ]: `\u200e@${ formFields.domain.value }\u202c`,
-	};
+type MailboxFormWrapperWithCommonProps = MailboxFormWrapperProps & {
+	getCommonFieldProps: ReturnType< typeof createCommonFieldPropsHandler >;
+	formFields: MailboxFormFields;
+};
 
+const createCommonFieldPropsHandler = ( formIndex: number, isAutoFocusEnabled: boolean ) => {
```

Comment:
Do we use this higher order function to support the `ReturnType< typeof createCommonFieldPropsHandler >` type definition? It might improve readability to inline it in `MailboxFormWrapper`, but if it's for the sake of that succinct type definition, I can see the point.

Reply:
Now inlined 👍

Reply:
My thinking here wasn't that the type definition should be inlined, but rather that the returned function from `createCommonFieldPropsHandler` could perhaps be inlined in the `MailboxFormWrapper` component directly. Why? Because the higher order function makes no difference to how the code runs (from what I could gather) but it reduces the readability of the code IMO.

But if this complicates the type annotations, it might not be worth it.